### ISSUE TITLE
Accessibility focus issues with the modal

### ISSUE THREAD

**Issue by roebuk** [everfund/everfund#64]
**Describe the bug**
When the Everfund modal is activated the users focus should be brough into the modal, whilst the modal is open, the users keyboard focus should be unable to escape contents of the modal. Upon close of the modal the focus should be restored to button that activated the modal.

**To Reproduce**
Steps to reproduce the behaviour:

1. Open the model, the focus remains on the button that activate the modal.
2. Tab around and eventually you'll be able to tab outside of the modal.
3. Close the modal and you're focus isn't taken back the activating button.

**Expected behavior**
The focus behaviour should match the [WAI-ARIA authoring practices on modals](https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html)
